code/modules/integrated_electronics/core/integrated_circuit.dm

```diff
@@ -0,0 +1,424 @@
+	var/extended_desc
+	var/list/inputs = list()
+	var/list/inputs_default = list()			// Assoc list which will fill a pin with data upon creation.  e.g. "2" = 0 will set input pin 2 to equal 0 instead of null.
+	var/list/outputs = list()
+	var/list/outputs_default = list()		// Ditto, for output.
+	var/list/activators = list()
+	var/next_use = 0 				//Uses world.time
+	var/complexity = 1 				//This acts as a limitation on building machines, more resource-intensive components cost more 'space'.
+	var/size						//This acts as a limitation on building machines, bigger components cost more 'space'. -1 for size 0
+	var/cooldown_per_use = 9 // Circuits are limited in how many times they can be work()'d by this variable.
+	var/power_draw_per_use = 0 		// How much power is drawn when work()'d.
+	var/power_draw_idle = 0			// How much power is drawn when doing nothing.
+	var/spawn_flags					// Used for world initializing, see the #defines above.
+	var/category_text = "NO CATEGORY THIS IS A BUG"	// To show up on circuit printer, and perhaps other places.
+	var/removable = TRUE 			// Determines if a circuit is removable from the assembly.
+	var/displayed_name = ""
+	var/allow_multitool = 1			// Allows additional multitool functionality
+									// Used as a global var, (Do not set manually in children).
+/proc/initialize_integrated_circuits_list()
+	for(var/thing in typesof(/obj/item/integrated_circuit))
+		GLOB.all_integrated_circuits += new thing()
+
+/*
+	Integrated circuits are essentially modular machines.  Each circuit has a specific function, and combining them inside Electronic Assemblies allows
+a creative player the means to solve many problems.  Circuits are held inside an electronic assembly, and are wired using special tools.
+*/
+
+/obj/item/integrated_circuit/examine(mob/user)
+	interact(user)
+	external_examine(user)
+	. = ..()
+
+// This should be used when someone is examining while the case is opened.
+/obj/item/integrated_circuit/proc/internal_examine(mob/user)
+	to_chat(user, "This board has [inputs.len] input pin\s, [outputs.len] output pin\s and [activators.len] activation pin\s.")
+	for(var/datum/integrated_io/I in inputs)
+		if(I.linked.len)
+			to_chat(user, "The '[I]' is connected to [I.get_linked_to_desc()].")
+	for(var/datum/integrated_io/O in outputs)
+		if(O.linked.len)
+			to_chat(user, "The '[O]' is connected to [O.get_linked_to_desc()].")
+	for(var/datum/integrated_io/activate/A in activators)
+		if(A.linked.len)
+			to_chat(user, "The '[A]' is connected to [A.get_linked_to_desc()].")
+	any_examine(user)
+	interact(user)
+
+// This should be used when someone is examining from an 'outside' perspective, e.g. reading a screen or LED.
+/obj/item/integrated_circuit/proc/external_examine(mob/user)
+	any_examine(user)
+
+/obj/item/integrated_circuit/proc/any_examine(mob/user)
+	return
+
+/obj/item/integrated_circuit/New()
+	displayed_name = name
+	if(!size) size = w_class
+	if(size == -1) size = 0
```

Comment:
These 2 lines shouldn't be necessary

Also stop doing 1 line if blocks

Reply:
no,it's neccesary. It helps to avoid bugs

Reply:
No it really isn't. If the default values have been modified whoever modified them incorrectly has to deal with the problem themselves, not get saved by excessive checks.